utils/wasm-builder/src/cargo_command.rs

```diff
@@ -135,10 +135,37 @@ impl CargoCommand {
         Ok(())
     }
 
-    fn remove_cargo_encoded_rustflags(&self, command: &mut Command) {
-        // substrate's wasm-builder removes these vars so do we
-        // check its source for details
-        command.env_remove("CARGO_ENCODED_RUSTFLAGS");
-        command.env_remove("RUSTFLAGS");
+    fn clean_up_environment(&self, command: &mut Command) {
+        // Inherited build scripts environment variables must be removed,
+        // otherwise we will get compilation errors on the stable toolchain.
+
+        for env_var in [
+            "CARGO",
+            "CARGO_MANIFEST_DIR",
+            "CARGO_MANIFEST_LINKS",
+            "CARGO_MAKEFLAGS",
+            "OUT_DIR",
+            "TARGET",
+            "HOST",
+            "NUM_JOBS",
+            "OPT_LEVEL",
+            "PROFILE",
+            "RUSTC",
+            "RUSTDOC",
+            "RUSTC_WRAPPER",
+            "RUSTC_WORKSPACE_WRAPPER",
+            "RUSTC_LINKER",
```

Comment:
mb reconsider:

- `TARGET`,
- `OUT_DIR`
- `RUSTC_WRAPPER`

for cache managing,

- `OPT_LEVEL`,
- `PROFILE`

for syncing the original package?

Reply:
@clearloop `TARGET` will be passed using `.arg("--target=wasm32-unknown-unknown")`, it's not host target

This will be ignored by cargo and will not change behavior ([Environment variables Cargo reads](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-reads)):
- `OUT_DIR`
- `OPT_LEVEL`
- `PROFILE`

`RUSTC_WRAPPER` maybe make sence

Reply:
done in 89fc8998b77929c2f3431bca18031cfeedc4817e